### DISCUSSION TITLE
There is a problem with the size of the previewed image

### DISCUSSION THREAD

**Discussion by harisonkhlil** [sxyazi/yazi#820] [1. Q&A] [answered]
<img width="1090" alt="image" src="https://github.com/sxyazi/yazi/assets/70578017/a8965d4a-e3d3-422a-8c5e-1cc131571a5e">
<img width="693" alt="image" src="https://github.com/sxyazi/yazi/assets/70578017/de1e8e8f-4c29-41b7-aa5d-4a372f873bdf">
<img width="519" alt="image" src="https://github.com/sxyazi/yazi/assets/70578017/2e53f880-c9fa-49ba-b62a-47be8b59cc6f">

But your preview function is fine

![image](https://github.com/sxyazi/yazi/assets/70578017/af4d5c29-81d7-4029-a9d9-1af7c9b9e2fc)

**Comment by sxyazi** [comment]
See https://yazi-rs.github.io/docs/image-preview#why-wont-my-images-adapt-to-terminal-size

**Reply by harisonkhlil** [reply]
I didn't set `max_width` and `max_height`.
```toml
[preview]
# tab_size = 4
image_quality = 90
```
I'm using a kitty terminal and couldn't find the relevant `pixel` setting

**Reply by sxyazi** [reply, marked as answer]
You need to tune the [`max_width` and `max_height`](https://yazi-rs.github.io/docs/configuration/yazi#preview) up as what you want - which will fit your terminal automatically, that's necessary for [pre-caching images](https://yazi-rs.github.io/blog#pre-caching)
